hudi-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadata.java

```diff
@@ -686,4 +597,26 @@ record -> {
         .stream()
         .collect(Collectors.groupingBy(Pair::getKey, Collectors.mapping(Pair::getValue, Collectors.toSet())));
   }
+
+  /**
+   * Derive necessary properties for FG reader.
+   */
+  TypedProperties buildFileGroupReaderProperties(HoodieMetadataConfig metadataConfig) {
+    HoodieCommonConfig commonConfig = HoodieCommonConfig.newBuilder()
+        .fromProperties(metadataConfig.getProps()).build();
+    TypedProperties props = TypedProperties.copy(metadataConfig.getProps());
```

Comment:
the `metadataConfig.getProps()` include all the write config options, it's more safe we just set up the options required.